napari/utils/_pydantic/evented_model.py

```diff
@@ -0,0 +1,67 @@
+from typing import Dict
+
+from pydantic import BaseModel, PrivateAttr
+
+from ..events.dataclass import _type_to_compare, is_equal
+from ..events.event import EmitterGroup
+from .custom_types import JSON_ENCODERS
+
+
+class EventedModel(BaseModel):
+
+    # add private attributes for event emission
+    _events: EmitterGroup = PrivateAttr(default_factory=EmitterGroup)
+    __equality_checks__: Dict = PrivateAttr(default_factory=dict)
+
+    # Definte the config so that assigments are validated
+    # and add custom encoders
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+        underscore_attrs_are_private = True
+        use_enum_values = True
+        validate_all = True
+        json_encoders = JSON_ENCODERS
```

Comment:
```suggestion
        # whether to allow arbitrary user types for fields (they are validated
        # simply by checking if the value is an instance of the type). If
        # False, RuntimeError will be raised on model declaration
        arbitrary_types_allowed = True
        # whether to perform validation on assignment to attributes
        validate_assignment = True
        # whether to treat any underscore non-class var attrs as private
        # https://pydantic-docs.helpmanual.io/usage/models/#private-model-attributes
        underscore_attrs_are_private = True
        # whether to populate models with the value property of enums, rather
        # than the raw enum. This may be useful if you want to serialise
        # model.dict() later
        use_enum_values = True
        # whether to validate field defaults (default: False)
        validate_all = True
        # a dict used to customise the way types are encoded to JSON
        # https://pydantic-docs.helpmanual.io/usage/exporting_models/#modeljson
        json_encoders = JSON_ENCODERS
```

Reply:
lets just comment the crap out of EventedModel, so we don't have to go to pydantic docs each time